mongo/options/sessionoptions.go

```diff
@@ -100,6 +110,12 @@ func MergeSessionOptions(opts ...*SessionOptions) *SessionOptions {
 		if opt.DefaultMaxCommitTime != nil {
 			s.DefaultMaxCommitTime = opt.DefaultMaxCommitTime
 		}
+		if opt.Snapshot != nil {
+			s.Snapshot = opt.Snapshot
+		}
+	}
+	if s.CausalConsistency == nil && (s.Snapshot == nil || !*s.Snapshot) {
```

Comment:
The condition `s.Snapshot == nil` relies on the default value for an unset `Snapshot` being `false`, but would become incorrect if we change the default value for an unset `Snapshot` to `true`. Does it make sense to encode that default value condition into a function so we can change it in one place?

E.g.
```go
func (sopts *SessionOptions) IsSnapshot() bool {
    return sopts.Snapshot != nil && s.Snapshot
}
```

Reply:
This is the only place where we have to check if its nil or false (the non-options code just copies the value regardless of what it its) so I'm not sure how much we would benefit from that. Also, because causalConsistency and Snapshot are mutually exclusive, changing Snapshot to default to true would functionally require us to stop defaulting causalConsistency to true.